repo/modules.go

```diff
@@ -81,6 +88,9 @@ func importRepoRulesModules(filename string, _ *RemoteCache) (repos []Repo, err
 		if mod, ok := pathToModule[path]; ok {
 			mod.Sum = sum
 		}
+		if mod, ok := replacePathToModule[path]; ok {
+			mod.Sum = sum
+		}
 	}
 
 	// If sums are missing, run go mod download to get them.
```

Comment:
If the sum is missing for a module with a replace directive, then we should use `replace.Path` and `replace.Version` to retrieve the sum via `goModDownload`.

Reply:
Yep makes sense and this sort of comes about naturally once I removed replacePathToModule

Reply:
We also need to modify `missingSumArgs` to contain `mod.Replace.Path` and `mod.Replace.Version` if `mod.Replace != nil`

Reply:
:+1: